### PR TITLE
EXAMPLES: moved examples to root

### PR DESCRIPTION
**Pull request by hoopoepg**
- examples are moved into root directory
- updated tests
- updated rpmbuild and pkgconfig scripts

### CODE REVIEW

File: examples/Makefile.am [hunk 1]
```diff
@@ -6,11 +6,11 @@
 #
 
 examplesdir = $(pkgdatadir)/examples
-dist_examples_DATA = \
+dist_examples_DATA =  \
 	ucx_hello_world.h \
 	ucp_hello_world.c \
 	uct_hello_world.c \
-	ucx_profiling.c \
+	ucx_profiling.c   \
```
Comment (yosefe) [inline examples/Makefile.am:13]: let's move ucx_profiling to test/apps, since it's not really an example

File: examples/Makefile.am [hunk 2]
```diff
@@ -6,11 +6,11 @@
 #
 
 examplesdir = $(pkgdatadir)/examples
-dist_examples_DATA = \
+dist_examples_DATA =  \
```
Comment (yosefe) [inline examples/Makefile.am:9]: maybe rename "ucx_hello_world.{h,c}" to "hello_world_common.{h,c}" ?

File: examples/Makefile.am (hunk 1 above)
Comment (hoopoepg) [reply examples/Makefile.am]: moved

File: examples/Makefile.am (hunk 2 above)
Comment (hoopoepg) [reply examples/Makefile.am:9]: there is no ucx_hello_world.**c** file, and there are hello world sources:
```ucp_hello_world.c  uct_hello_world.c  ucx_hello_world.h```
renaming header file to ```hello_world_common.h``` will look strange:
```ucp_hello_world.c  uct_hello_world.c hello_world_common.h```

what do you think?

Comment (yosefe) [reply examples/Makefile.am:9]: i suggest something like:
ucp_hello_world.c uct_hello_world.c hello_world_common.h
or
ucp_hello_world.c uct_hello_world.c common.h
or
ucp_hello_world.c uct_hello_world.c hello_world_util.h

Comment (hoopoepg) [reply examples/Makefile.am]: ok, done

**Pull request merged**